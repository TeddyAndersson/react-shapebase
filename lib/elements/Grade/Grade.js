import React from 'react';
import _ from 'lodash';
import styled, { css } from 'styled-components'
import PropTypes from 'prop-types';
import SquareShape from '../Shape/SquareShape';

class Element extends React.Component{
  constructor(props){
    super(props);
    this.state = {
      activeOption: this.props.initialGrade,
    }
  }

  handleClick = (id) => {
    this.setState({activeOption: id})

    _.invoke(this.props, 'onClick', id, this.props)
  }

  render(){
    const {
        className,
        range,
    } = this.props

    const options = [...Array(range+1).keys()].slice(1)

    return (
        <ul className={className}>
          { _.values(options).map(option => (
            // Replace <li/> with a component for better state handling and generic code
                <li
                    key={option}
                    onClick={() => this.handleClick( option )}
                >
                    <SquareShape
                        background={`${ option === this.state.activeOption ? option.background : 'rgba(0,0,0,0.1)'}`}
                        raised={option === this.state.activeOption ? 1 : 0}
                    >
                        <input
                        type="hidden"
                        name={option}
                        value={option}
                        autoComplete="off"
                        />
                        <div>
                            <span>
                                {option}
                            </span>
                        </div>
                    </SquareShape>
                </li>
            ))}
        </ul>
    )
  };
};

Element.propTypes = {

    /**
      * The children of the component.
      */
    children: PropTypes.node,

    /**
      * Additonal classes.
      */
    className: PropTypes.string,

    /**
     * Description of the intreval the range.
     */
    description: PropTypes.bool,

    /**
     * The range of the grades options 1-5 or 1-10
     */
    range: PropTypes.oneOf([5,10]),

    /**
     * The range of the grades options 1-5 or 1-10
     */
    initialGrade: PropTypes.number,
}

Element.defaultProps = {
    range: 5,
    initialGrade: undefined,
}

const Grade = styled(Element)`
    display: flex;
    justify-content: space-between;
    padding: 0;
    list-style: none;
    display: flex;
    margin: 0;
    font-weight: bold;
    li {
        div {
            display: flex;
            justify-content: center;
            align-items: center;
            transition: all 0.3s ease
        }
    }
    ${props => props.range === 10 && css`
        font-size: 0.6em;
    `}
`;

export default Grade;