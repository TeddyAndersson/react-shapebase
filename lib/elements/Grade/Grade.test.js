import React from 'react';
import Grade from './Grade'
import renderer from 'react-test-renderer';

test('should render Grade', () => {
  const component = renderer.create(
    <Grade
      range={10}
    />
  );
  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
})
