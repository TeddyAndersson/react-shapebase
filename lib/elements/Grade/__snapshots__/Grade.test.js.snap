// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Grade 1`] = `
<ul
  className="sc-htpNat hHKzqV"
>
  <li
    onClick={[Function]}
  >
    <div
      className="sc-bwzfXH cRQsEN sc-bdVaJa jmjnGv"
    >
      <input
        autoComplete="off"
        name={1}
        type="hidden"
        value={1}
      />
      <div>
        <span>
          1
        </span>
      </div>
    </div>
  </li>
  <li
    onClick={[Function]}
  >
    <div
      className="sc-bwzfXH cRQsEN sc-bdVaJa jmjnGv"
    >
      <input
        autoComplete="off"
        name={2}
        type="hidden"
        value={2}
      />
      <div>
        <span>
          2
        </span>
      </div>
    </div>
  </li>
  <li
    onClick={[Function]}
  >
    <div
      className="sc-bwzfXH cRQsEN sc-bdVaJa jmjnGv"
    >
      <input
        autoComplete="off"
        name={3}
        type="hidden"
        value={3}
      />
      <div>
        <span>
          3
        </span>
      </div>
    </div>
  </li>
  <li
    onClick={[Function]}
  >
    <div
      className="sc-bwzfXH cRQsEN sc-bdVaJa jmjnGv"
    >
      <input
        autoComplete="off"
        name={4}
        type="hidden"
        value={4}
      />
      <div>
        <span>
          4
        </span>
      </div>
    </div>
  </li>
  <li
    onClick={[Function]}
  >
    <div
      className="sc-bwzfXH cRQsEN sc-bdVaJa jmjnGv"
    >
      <input
        autoComplete="off"
        name={5}
        type="hidden"
        value={5}
      />
      <div>
        <span>
          5
        </span>
      </div>
    </div>
  </li>
  <li
    onClick={[Function]}
  >
    <div
      className="sc-bwzfXH cRQsEN sc-bdVaJa jmjnGv"
    >
      <input
        autoComplete="off"
        name={6}
        type="hidden"
        value={6}
      />
      <div>
        <span>
          6
        </span>
      </div>
    </div>
  </li>
  <li
    onClick={[Function]}
  >
    <div
      className="sc-bwzfXH cRQsEN sc-bdVaJa jmjnGv"
    >
      <input
        autoComplete="off"
        name={7}
        type="hidden"
        value={7}
      />
      <div>
        <span>
          7
        </span>
      </div>
    </div>
  </li>
  <li
    onClick={[Function]}
  >
    <div
      className="sc-bwzfXH cRQsEN sc-bdVaJa jmjnGv"
    >
      <input
        autoComplete="off"
        name={8}
        type="hidden"
        value={8}
      />
      <div>
        <span>
          8
        </span>
      </div>
    </div>
  </li>
  <li
    onClick={[Function]}
  >
    <div
      className="sc-bwzfXH cRQsEN sc-bdVaJa jmjnGv"
    >
      <input
        autoComplete="off"
        name={9}
        type="hidden"
        value={9}
      />
      <div>
        <span>
          9
        </span>
      </div>
    </div>
  </li>
  <li
    onClick={[Function]}
  >
    <div
      className="sc-bwzfXH cRQsEN sc-bdVaJa jmjnGv"
    >
      <input
        autoComplete="off"
        name={10}
        type="hidden"
        value={10}
      />
      <div>
        <span>
          10
        </span>
      </div>
    </div>
  </li>
</ul>
`;
